src/lib/kdf/kdf2/kdf2.cpp

```diff
@@ -11,7 +11,8 @@ namespace Botan {
 
 size_t KDF2::kdf(byte key[], size_t key_len,
                  const byte secret[], size_t secret_len,
-                 const byte salt[], size_t salt_len) const
+                 const byte salt[], size_t salt_len,
+                 const byte[], size_t) const
```

Comment:
I dislike the label being ignored entirely here. An application might reasonably assume that KDF(K,label1) != KDF(K,label2)

The label should be consumed along with the salt, say for concreteness taking the label before the salt. For implementations that don't support a label input an identical KDF output is still easily derived.